File: CLOMI/workshop/ressources/annexe/python/Kafka/producer.py
from kafka import KafkaProducer
import json

# le message est un dict, cela veut dire { key: value, ... }
def KafkaSendMessage(topic: str, message: dict):
    producer = KafkaProducer(
        bootstrap_servers='localhost:9092',
        value_serializer=lambda v: json.dumps(v).encode('utf-8')
    )
    producer.send(topic, value=message)
    producer.flush()

File: CLOMI/workshop/ressources/annexe/node/Kafka/producer.js
import { Kafka } from 'kafkajs'

const kafka = new Kafka({
    clientId: "app",
    brokers: ['localhost:9092', 'localhost:9092']
})

const producer = kafka.producer()

await producer.connect();

export const KafkaSendMessage = async (topicName, message) => {
    if (typeof(message) == "string")
        producer.send({
            topic: topicName,
            messages: [
                { value: message }
            ]
        });
    else if (typeof(message) == "object")
        producer.send({
            topic: topicName,
            messages: [
                message.forEach(element => {value: element})
            ]
        });
    else
        console.error("Message must be a string or a list")
}
